mediapipe/tasks/ios/vision/hand_landmarker/sources/MPPHandLandmarker.mm

```diff
@@ -0,0 +1,288 @@
+
+#import "mediapipe/tasks/ios/common/utils/sources/MPPCommonUtils.h"
+#import "mediapipe/tasks/ios/common/utils/sources/NSString+Helpers.h"
+#import "mediapipe/tasks/ios/core/sources/MPPTaskInfo.h"
+#import "mediapipe/tasks/ios/vision/core/sources/MPPVisionPacketCreator.h"
+#import "mediapipe/tasks/ios/vision/core/sources/MPPVisionTaskRunner.h"
+#import "mediapipe/tasks/ios/vision/hand_landmarker/sources/MPPHandLandmarksConnections.h"
+#import "mediapipe/tasks/ios/vision/hand_landmarker/utils/sources/MPPHandLandmarkerOptions+Helpers.h"
+#import "mediapipe/tasks/ios/vision/hand_landmarker/utils/sources/MPPHandLandmarkerResult+Helpers.h"
+
+namespace {
+using ::mediapipe::NormalizedRect;
+using ::mediapipe::Packet;
+using ::mediapipe::Timestamp;
+using ::mediapipe::tasks::core::PacketMap;
+using ::mediapipe::tasks::core::PacketsCallback;
+}  // namespace
+
+static NSString *const kImageTag = @"IMAGE";
+static NSString *const kImageInStreamName = @"image_in";
+static NSString *const kNormRectTag = @"NORM_RECT";
+static NSString *const kNormRectInStreamName = @"norm_rect_in";
+static NSString *const kImageOutStreamName = @"image_out";
+static NSString *const kLandmarksTag = @"LANDMARKS";
+static NSString *const kLandmarksOutStreamName = @"hand_landmarks";
+static NSString *const kWorldLandmarksTag = @"WORLD_LANDMARKS";
+static NSString *const kWorldLandmarksOutStreamName = @"world_hand_landmarks";
+static NSString *const kHandednessTag = @"HANDEDNESS";
+static NSString *const kHandednessOutStreamName = @"handedness";
+static NSString *const kTaskGraphName =
+    @"mediapipe.tasks.vision.hand_landmarker.HandLandmarkerGraph";
+static NSString *const kTaskName = @"handLandmarker";
+
+#define InputPacketMap(imagePacket, normalizedRectPacket)   \
+  {                                                         \
+    {kImageInStreamName.cppString, imagePacket}, {          \
+      kNormRectInStreamName.cppString, normalizedRectPacket \
+    }                                                       \
+  }
+
+@interface MPPHandLandmarker () {
+  /** iOS Vision Task Runner */
+  MPPVisionTaskRunner *_visionTaskRunner;
+  dispatch_queue_t _callbackQueue;
+}
+@property(nonatomic, weak) id<MPPHandLandmarkerLiveStreamDelegate> handLandmarkerLiveStreamDelegate;
+@end
+
+@implementation MPPHandLandmarker
+
+- (nullable MPPHandLandmarkerResult *)handLandmarkerResultWithOutputPacketMap:
+    (PacketMap &)outputPacketMap {
+  return [MPPHandLandmarkerResult
+      handLandmarkerResultWithLandmarksPacket:outputPacketMap[kLandmarksOutStreamName.cppString]
+                         worldLandmarksPacket:outputPacketMap[kWorldLandmarksOutStreamName
+                                                                  .cppString]
+                             handednessPacket:outputPacketMap[kHandednessOutStreamName.cppString]];
+}
+
+- (void)processLiveStreamResult:(absl::StatusOr<PacketMap>)liveStreamResult {
+  if (![self.handLandmarkerLiveStreamDelegate
```

Comment:
FYI I got feedback in an earlier CL to make the `didFinishDetectionWithResult` method in the protocol non-optional and instead make the delegate optional. This is meant to make implementing the delegate less error-prone. Can you update the PR?

Reply:
The delegate is already optional. But we don't let the task get initialised if user doesn't set it in the `MPPVisionTaskRunner`. I'll raise separate PRs for this since it will affect all tasks and the tests too.

The method is optional because in future we will also add a method with an extra argument for returning the source MPPImage. So users can implement either one of them based on what they prefer. Even if I change it now, it may end up getting updated later.
Not uncommon to have protocols with only optionals methods. The following protocol delivers camera buffers in iOS.
https://developer.apple.com/documentation/avfoundation/avcapturevideodataoutputsamplebufferdelegate?language=objc

LMK what you think.

Reply:
Also when you remove the mandatory delegate check we have either one of these options:
1. Provide a C++ packets callback irrespective. Otherwise users won't be able to call `detectAsyncInImage()` without facing callback errors from C++.
2. Not provide a packets callback and store a flag in vision task runner to indicate the same. Use this flag to return a better error message to users when `processLiveStreamPacketMap()` is called since the C++ error would not be meaningful to the iOS clients.
I prefer option 2 as this would avoid any unnecessary processing.

Reply:
Thanks for the explanation. Do you think it makes sense to have two different delegate protocols, each with their own non-optional method? This might be more cumbersome than having a single delegate with optional methods but I just wanted to raise the possibility. The reviewer was concerned that there is very little compile time safety for optional protocols (e.g. if a user mistypes a method name, nothing happens). I will defer to your judgment here. Please do update the FaceLandmarker and FaceDetector APIs as well so we have consistency.

As for your last question, I have a strong preference for option 2.